src/ToastNotification/TimedToastNotification.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import helpers from '../common/helpers'
+import Timer from '../common/Timer'
+import { ToastNotification } from '../index.js'
+
+/**
+ * TimedToastNotification Component for Patternfly React
+ */
+class TimedToastNotification extends React.Component {
+  constructor(props) {
+    super(props)
+    helpers.bindMethods(this, ['onMouseEnter', 'onMouseLeave'])
+  }
+
+  componentDidMount() {
+    const { onDismiss, timerDelay } = this.props
+
+    this.timer = new Timer(onDismiss, timerDelay)
+    this.timer.startTimer()
+  }
+
+  componentWillUnmount() {
+    this.timer && this.timer.clearTimer()
+  }
+
+  onMouseEnter() {
+    this.timer && this.timer.clearTimer()
+  }
+
+  onMouseLeave() {
+    this.timer && this.timer.startTimer()
+  }
+  render() {
+    const { notificationClass, type, onDismiss, children } = this.props
+    return (
```

Comment:
If I understand it correctly, you'll need to wrap this in additional div and give it onMouseEnter and onMouseLeave props. Otherwise those won't ever get triggered.

Also we'll need to think about this further as notification pausing should happen for all notifications in the 'notification toaster' - a container of all toast notification. Because in case when 3 notification appear at once, user is not able to pause them all at once by hovering.